### PR TITLE
Soloninko Andrey. Lab №1, var 1

### PR DESCRIPTION
**Pull request by Manymoera**
Implemented a plugin that outputs name of users classes and their fields.
Example: 
`Struct MyClass{
     int a;
     int b;
}`
will be outputed as:
`MyClass
     |_a
     |_b
`

### CODE REVIEW

File: clang/test/lab1/soloninko_andrey/ClassListTests.cpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    int variable;
+};
+
+// CHECK: MyClass1
+class MyClass1 {
+    // CHECK-NEXT: |_var
+    float var;
+    // CHECK-NEXT: |_var1
+    double var1;
+};
```
Comment (m-ly4) [inline clang/test/lab1/soloninko_andrey/ClassListTests.cpp:29]: Could you please add a test with class definition inside another class?

File: clang/test/lab1/soloninko_andrey/ClassListTests.cpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    // CHECK-NEXT: |_variable
+    int variable;
+};
+
+// CHECK: MyClass1
+class MyClass1 {
+    // CHECK-NEXT: |_var
+    float var;
+    // CHECK-NEXT: |_var1
+    double var1;
```
Comment (m-ly4) [inline clang/test/lab1/soloninko_andrey/ClassListTests.cpp:17]: Could you please add a test with template class declaration?

File: clang/labs/lab1/soloninko_andrey/classList.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+#include "llvm/Support/raw_ostream.h"
+
+using namespace clang;
```
Comment (aobolensk) [inline clang/labs/lab1/soloninko_andrey/classList.cpp]: Please, avoid using "using namespace *"

**Comment by Manymoera** [comment]
More cases were added for tests such as: 
1. Class definition inside another class
2. Template class declaration

Also deleted `using namespace clang`

**Review by m-ly4 — APPROVED** [review]

**Review by aobolensk — APPROVED** [review]
LGTM

**Pull request merged**